### ISSUE TITLE
DockerComposeEnvironment.up() does not fail when initial connection to Ryuk fails

### ISSUE THREAD

**Issue by mircohaug** [testcontainers/testcontainers-node#411]
I am using the colima docker wrapper on Mac and this engine is a little bit slower to open ports on the host machine as is Docker Desktop. 

**Expected Behaviour**
When testcontainers can not connect to ryuk it should either retry the connection after some offset or fail the `.up()` command.

**Actual Behaviour**
When testcontainers fails to connect to ryuk a log message is written but otherwise nothing happens. Also the `.withStartupTimeout(timeout)` does not fire. The command runs until the job reaches the execution time limit and is killed. A short time after testcontainers stops doing anything, I can connect to Ryuk using `nc`. This error does not happen all the time. It seems like this is a race condition.

**Testcontainer Logs**
```
  testcontainers INFO  Starting DockerCompose environment: testcontainers-541a6b7adf8a1a701476c2e6112d5eef +0ms
  testcontainers DEBUG Creating new Reaper for session: 95974014f35659df4e3b688638658b13 +1ms
  testcontainers DEBUG Found applicable Docker client strategy: UnixSocketStrategy +2ms
  testcontainers DEBUG Testing Docker client strategy URI: unix:///var/run/docker.sock +0ms
  testcontainers INFO  Using Docker client strategy: UnixSocketStrategy, Docker host: localhost +13ms
  testcontainers DEBUG Fetching system diagnostics +1ms
  testcontainers DEBUG Found applicable registry auth locator for registry "https://index.docker.io/v1/": Auths +3ms
  testcontainers DEBUG Checking if image exists: testcontainers/ryuk:0.3.2 +1ms
  testcontainers DEBUG Not pulling image as it already exists: testcontainers/ryuk:0.3.2 +73ms
  testcontainers INFO  Creating container for image: testcontainers/ryuk:0.3.2 +2ms
  testcontainers DEBUG System diagnostics: {"node":{"version":"v16.14.2","architecture":"x64","platform":"darwin"},"docker":{"serverVersion":"20.10.11","operatingSystem":"Alpine Linux v3.14","operatingSystemType":"linux","architecture":"x86_64","cpus":6,"memory":6234644480},"dockerCompose":{"version":"2.12.2"}} +37ms
  testcontainers INFO  Starting container testcontainers/ryuk:0.3.2 with ID: 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5 +100ms
  testcontainers:containers TRACE 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5: 2022/11/29 07:16:25 Pinging Docker... +0ms
  testcontainers:containers TRACE 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5: 2022/11/29 07:16:25 Docker daemon is available! +0ms
  testcontainers:containers TRACE 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5: 2022/11/29 07:16:25 Starting on port 8080... +0ms
  testcontainers:containers TRACE 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5: 2022/11/29 07:16:25 Started! +0ms
  testcontainers DEBUG Waiting for container to be ready: 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5 +743ms
  testcontainers DEBUG Waiting for log message "/.+ Started!/" +1ms
  testcontainers INFO  Container is ready +10ms
  testcontainers DEBUG Connecting to Reaper 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5 on localhost:49164 +1ms
  testcontainers ERROR Reaper 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5 socket error: Error: connect ECONNREFUSED 127.0.0.1:49164 +4ms
  testcontainers ERROR Connection to Reaper 7f8b43d514b412eca367bb1f007c2265d6a7cbc24978c1f925af60729fb97ff5 closed with error +1ms
```

**Steps to Reproduce**
1. In [this line](https://github.com/testcontainers/testcontainers-node/blob/08454582aee7efd74312236040dc76d6c907d3a5/src/reaper.ts#L126) changing the port to some other port should also create a Connection refused error on socket connect. The same as in the logs.

**Environment Information**
- Operating System: Docker on Mac using Colima
- Docker Version: 20.10.11
- Node version: v16.14.2
- Testcontainers version: 9.0.0

**Comment by cristianrgreco** [comment]
Thanks for raising @mircohaug. Raising a PR which will both retry the connection to the Reaper if it fails, and correctly throw if it fails to do so.

**Comment by mircohaug** [comment]
@cristianrgreco Thank you very much for fixing this ❤️ I was just about to start drafting a PR but your solution is ways better 😉 Can you given an ETA when this gets released?

**Comment by cristianrgreco** [comment]
@mircohaug released now in 9.0.1 🙂

**Comment by cristianrgreco** [comment]
@mircohaug FYI I am working on adding 1st class support for Colima here: https://github.com/testcontainers/testcontainers-node/pull/526. I've added a composite wait strategy to handle cases where the port is ready but colima has not registered it. So for example for the reaper we not only wait for the log message, but also for listening ports. As a result I don't believe this retry mechanism is necessary so I'm planning to remove it
